webapp/components/self-navigator.html

```diff
@@ -84,9 +84,10 @@
 
       /**
        * Get query params to persist when creating history.
+       * Defaults to the queryParams property.
        */
       navigationQueryParams() {
-        return {};
+        return Object.assign({}, this.queryParams);
```

Comment:
Just double-checking: Is `queryParams` a shallow object? Otherwise we may get some wonkiness here.

Reply:
Done.